tests/common/plugins/pdu_controller/__init__.py

```diff
@@ -35,3 +39,19 @@ def pdu_controller(duthosts, enum_rand_one_per_hwsku_hostname, conn_graph_facts,
     if controller:
         controller.turn_on_outlet()
         controller.close()
+
+@pytest.fixture(scope="module")
+def get_pdu_controller(conn_graph_facts, pdu):
+    controller = None
+
+    def pdu_controller_helper(duthost):
+        pdu_hosts = get_pdu_hosts(duthost)
+        controller = pdu_manager_factory(duthost.hostname, pdu_hosts, conn_graph_facts, pdu)
+
+        return controller
+
+    yield pdu_controller_helper
+
+    if controller:
```

Comment:
I _think_ this `controller` variable will not reflect any modification inside inner function.
The scope of `controller` inside inner function will start and end with the inner function, UNLES `controller` is a list in outer, and inner `append()` to it. This works because list is mutable.

If what I think is correct, what you could do is to use list to contain value of controller. OR, use function attributes. I would prefer function attribute.

Reply:
Good catch. I've pushed a new commit which stores used controllers in a dictionary to account for multiple controllers being accessed/used.